File: app/src/main/java/com/example/installer/utils/ApiServiceUtil.java
package com.example.installer.utils;


import com.example.installer.api.ApiService;
import com.example.installer.entity.APKEntity;
import com.example.installer.entity.PackageEntity;
import com.example.installer.entity.Result;

import java.util.List;

import rx.Observable;

/**
 * Author:  ZhangTao
 * Date: 2018/3/30.
 */
public class ApiServiceUtil {
    private static RetrofitUtil retrofitUtil;
    private static ApiService mApiService;

    static {
        retrofitUtil = RetrofitUtil.getInstance();
        mApiService = retrofitUtil.createService(ApiService.class);
    }

    public static Observable<Result<List<PackageEntity>>> getApplicationList() {
        return mApiService.getApplicationList();
    }

    public static Observable<Result<List<APKEntity>>> getPackageList(String system_name, String application_id, String version_type, int pageIndex) {
        return mApiService.getPackageList(system_name, application_id, version_type, pageIndex);
    }
}

File: app/src/main/java/com/example/installer/utils/ApiServiceUtil.kt
package com.example.installer.utils

import com.example.installer.api.APIService
import com.example.installer.entity.PackageEntity
import com.example.installer.entity.ProductEntity
import com.example.installer.entity.Result
import rx.Observable

/**
@author zhangtao
@date   2020/11/3

 **/
class ApiServiceUtil {
    companion object {
        private val mApiService by lazy {
            RetrofitUtil.getInstance().createService(APIService::class.java)
        }

        fun getProductList(): Observable<Result<List<ProductEntity>>> {
            return mApiService.getProductList()
        }

        fun getPackageList(
            system_name: String?,
            application_id: String?,
            version_type: String?,
            pageIndex: Int
        ): Observable<Result<List<PackageEntity>>> {
            return mApiService.getPackageList(
                system_name,
                application_id,
                version_type,
                pageIndex
            )
        }
    }
}